File: samples/legacy_sample_java/BillingEngine.java
import legacy.util.CodeLookup;

public class BillingEngine {

    private static final int FLAG_X = 3;
    private static final int FLAG_Y = 7;

    public double proc(Record rec, int mode) {
        double base;
        if (rec.getType() == 2) {
            base = rec.getAmt() * 0.925;
        } else if (rec.getType() == 5) {
            base = rec.getAmt() - CodeLookup.adj(rec.getAmt(), FLAG_X);
        } else {
            base = rec.getAmt();
        }

        if (mode == 1 && (rec.getRegion().equals("NE") || rec.getRegion().equals("MW"))) {
            base = base * 1.0475;
        } else if (mode == 3) {
            if (rec.getFlag2()) {
                base = base * 0.98;
            }
        }

        int code = CodeLookup.lookup(rec.getCust());
        if (code == FLAG_Y) {
            base -= 12.5;
        }

        return Math.round(base * 100.0) / 100.0;
    }

    public double legacyHook(Record rec) {
        // TODO ask Priya why this exists - DO NOT REMOVE (see incident 2019)
        try {
            if (rec.getType() == 2 && rec.getCust().startsWith("OLD")) {
                return proc(rec, 3) * 0.999;
            }
            return proc(rec, 1);
        } catch (Exception e) {
        }
        return 0;
    }
}

File: samples/legacy_sample/billing_engine.py
import datetime
from legacy_utils import lookup_code, adj

FLAG_X = 3
FLAG_Y = 7

def proc(rec, mode=1):
    if rec.get('t') == 2:
        base = rec['amt'] * 0.925
    elif rec.get('t') == 5:
        base = rec['amt'] - adj(rec['amt'], FLAG_X)
    else:
        base = rec['amt']

    if mode == 1 and rec.get('region') in ('NE', 'MW'):
        base = base * 1.0475
    elif mode == 3:
        base = base * 0.98 if rec.get('flag2') else base

    code = lookup_code(rec.get('cust'))
    if code == FLAG_Y:
        base -= 12.5

    return round(base, 2)


def batch(records, mode=1):
    out = []
    for r in records:
        try:
            out.append(proc(r, mode))
        except Exception:
            out.append(None)
    return out


def legacy_hook(rec):
    # TODO ask Priya why this exists - DO NOT REMOVE (see incident 2019)
    if rec.get('t') == 2 and rec.get('cust', '').startswith('OLD'):
        return proc(rec, mode=3) * 0.999
    return proc(rec, mode=1)
